src/main/java/Main.java

```diff
@@ -1,2 +1,24 @@
+import java.sql.SQLException;
+
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+
 public class Main {
+  public static void main(String[] args) throws SQLException {
+    String userName = args[0];
+    String password = args[1];
+    String urlPort = args[2];
+    String database = args[3];
+
+    DBDriverWrapper driver = new PostgresDriverWrapper(urlPort, database, userName, password);
+    VectorSchemaRoot vectorSchemaRoot = driver.executeQuery("select * from personaldata");
+    driver.getConnection().close();
```

Comment:
You are not closing the vectorSchemaRoot neither the allocators, please check it

Reply:
You can try using the autoclosable interface and then use the try with resources

Reply:
done